### PR TITLE
feat: add dynamic auth type

### PR DESCRIPTION
**Pull request by kycheng**
# Changes

- add dynamic define.
- add ListOption default pager.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] [`spec` PR link](https://github.com/katanomi/spec) included
- [x] Follows the [commit message standard](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md#commits)
- [x] Meets the [contributing guidelines](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md) (including
  functionality, content, code)
- [x] Test cases with documentation and functionality works as expected using current and related github repos (MUST deploy and check)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

### CODE REVIEW

File: apis/meta/v1alpha1/auth_types.go [hunk 1]
```diff
@@ -72,6 +73,8 @@ type AuthCheckStatus struct {
 	// AuthorizeURL url for the oAuth2 flow
 	// +optional
 	AuthorizeURL string `json:"authorizeURL,omitempty"`
+
+	AuthTokenStatus
```
Comment (kycheng) [inline apis/meta/v1alpha1/auth_types.go:77]: 用于动态密钥过期时间的返回

**Comment by kycheng** [comment]
lint 由于github 使用golang1.18 导致

File: apis/meta/v1alpha1/auth_types.go (hunk 1 above)
Comment (airycanon) [reply apis/meta/v1alpha1/auth_types.go:77]: 这里返回了 AuthTokenStatus 的 username 和 password 给前端，会有安全问题？

Comment (danielfbm) [reply apis/meta/v1alpha1/auth_types.go:77]: +1 感觉不应该直接返回所有的数据，是不是可以选择性挑需要的字段呢？

**Review by danielfbm — CHANGES_REQUESTED** [review]
除了评论的内容，看一下lint错误一起修复？

**Review by airycanon — APPROVED** [review]

File: apis/meta/v1alpha1/auth_types.go [hunk 2]
```diff
@@ -72,6 +73,10 @@ type AuthCheckStatus struct {
 	// AuthorizeURL url for the oAuth2 flow
 	// +optional
 	AuthorizeURL string `json:"authorizeURL,omitempty"`
+
+	// Provides the ability to convert username and password, and token refresh.
+	// +optional
+	ReflushData *AuthTokenStatus `json:"reflushData,omitempty"`
```
Comment (danielfbm) [inline apis/meta/v1alpha1/auth_types.go:79]: Reflush 是 `重新冲水` 的意思，应该是刷新？`RefreshData` ?

**Review by danielfbm — DISMISSED** [review]

**Review by danielfbm — APPROVED** [review]

**Pull request merged**